### PR TITLE
Lab4

### PR DESCRIPTION
**Pull request by Reqvel**
The fields in login page have been changed to CLIENT ID and CLIENT SECRET, authorization will be through Spotify. Music player beacame more interactive. Added side menu and logic for it with JS, optimized for small screen sizes. Added buttons listeners, interactive progressbar(using JS), playlists list generator, search page.

### CODE REVIEW

**Review by katetushkan — CHANGES_REQUESTED** [review]
Okay, I see the logo it's redirecting me to my account but then nothing happening. It's not redirecting after login, no playlists displayed in ur app, no songs even to try on smth, so what exactly u did in this work?

**Comment by Reqvel** [comment]
huh? What you mentioned is interaction with API and that's lab #5. And i specified in PR what i did for this lab. Or i dont get smth?

**Comment by katetushkan** [comment]
> huh? What you mentioned is interaction with API and that's lab #5. And i specified in PR what i did for this lab. Or i dont get smth?

After pressing login button, I'm redirecting to the Spotify, login to Spotify, and stays there. How am I supposed to continue use ur application if I'm on Spotify rn? what is CLIENT ID and CLIENT SECRET? as a user who opens ur app for the first time, where am I supposed to get this values?

**Comment by katetushkan** [comment]
Ohh, I see, I'm sorry, I thought u were trying to pass me lab5, I'm sorry that's my bad, a lot of u guys. Alright, sorry for confusing u, but anyway I'd like to have an answer to this
> as a user who opens ur app for the first time, where am I supposed to get this values?

I'll check 4th's work rn.

File: static/js/mp-menu.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+function showPlaylists(contentContainer, optContainer, optContainerClassname) {
+    // TODO
+    hideEl(optContainer, optContainerClassname)
+
+    contentContainer.innerHTML = '<div class="playlists-grid appear-animation"></div>'
+    const playlistsGrid = document.querySelector('.playlists-grid')
+
+    let gridItemsHTML = ''
+
+    for (let i = 30; i > 0; i--) {
```
Comment (katetushkan) [inline static/js/mp-menu.js:23]: Put it into constant, it makes ur code more readable.

**Review by katetushkan — APPROVED** [review]
Okay, it's fine, sorry once again for confusing u, `9` for u.

**Comment by Reqvel** [comment]
I am using an authorization type that does not require a server from my side. I will provide u with CLIENT ID and CLIENT SECRET. After that u'll sign in with ur spotify account. Maybe i can use another type, but this one for our needs is most suitable.

**Comment by katetushkan** [comment]
> I am using an authorization type that does not require a server from my side. I will provide u with CLIENT ID and CLIENT SECRET. After that u'll sign in with ur spotify account. Maybe i can use another type, but this one for our needs is most suitable.

OK.

**Review by katetushkan — APPROVED** [review]

**Pull request merged**